src/backend/distributed/commands/call.c

```diff
@@ -14,23 +14,23 @@
 
 #include "catalog/pg_proc.h"
 #include "commands/defrem.h"
-#include "distributed/citus_ruleutils.h"
-#include "distributed/colocation_utils.h"
+#include "distributed/deparser/citus_ruleutils.h"
+#include "distributed/utils/colocation_utils.h"
 #include "distributed/commands.h"
 #include "distributed/commands/multi_copy.h"
 #include "distributed/commands/utility_hook.h"
-#include "distributed/connection_management.h"
-#include "distributed/deparse_shard_query.h"
-#include "distributed/master_metadata_utility.h"
-#include "distributed/metadata_cache.h"
-#include "distributed/multi_executor.h"
-#include "distributed/multi_physical_planner.h"
-#include "distributed/adaptive_executor.h"
-#include "distributed/remote_commands.h"
-#include "distributed/shard_pruning.h"
+#include "distributed/connection/connection_management.h"
```

Comment:
In the code conventions, we sort the includes in the alphabetical order. We're breaking this now. Are we OK with that?

Reply:
Is there a value in having them in sorted? What can we infer by looking at the headers? Usually I navigate to the declaration of a method/variable and it opens the header automatically.

Reply:
I think it is useful for two things:

- it prevents adding the same header twice
- especially with your changes, the related headers will be next to each other. For example, we can see which headers under `connection` is added in a source file.
- (and a minor addition) I feel like the code is better structured with the strict ordering

Is there an easy way to sort them? Maybe adding such logic to `citus_indent` is better?

Reply:
https://github.com/TheOstrichIO/cpplint/tree/nitpick-sort-includes/cpplint this could be useful.

Reply:
I think this gets kind of tricky because:
- We need some headers to come before others, for example pg version constants are defined in a header(`distributed/pg_version_constants.h`) and that header should come before the usages of MACROS such as `PG_VERSION_12`.
- #if directives should also be sorted with the relative header, so it is not just sorting the #include lines.

Reply:
@JelteF do you know an easy way to sort them?

Reply:
According to @thanodnl, it's only a single uncrustify setting. I think he was talking about `mod_sort_include`:
https://github.com/uncrustify/uncrustify/blob/cf50d37987b083837005612cb315a7eb4e69d306/documentation/htdocs/options_ModifyCode.html#L122-L128

Reply:
There is also a trick for certain headers that need to be in a specific order (like having `#inlcude "postgres.h"` before anything else; group them in blocks.

uncrustify only sorts them when they are in the same block iirc. So by separating blocks with an empty line between them could help keeping them in order, the same for our pg11 and 12 specific includes.

the includes would look like:
```C
#include "postgres.h"

#include "catalog/..."
#include "distributed/..."
#include "nodes/..."

#if PGVERSION11_CHECK
#include "headers/specific/to/pg11"
#endif

#if PGVERSION12_CHECK
#include "headers/specific/to/pg12"
#endif
```

Hope this gives an idea.